src/pages/workspaces/workspace/_tests/WorkspaceMenu.test.js

```diff
@@ -0,0 +1,258 @@
+    }))
+    fireEvent.mouseOver(screen.getByText('Share'))
+    expect(screen.queryByText('Sharing is not currently supported on Azure Workspaces')).toBeNull()
+  })
+})
+
+
+describe('WorkspaceMenu - GCP workspace', () => {
+  beforeEach(() => {
+    useWorkspaceDetails.mockReturnValue({ workspace: { canShare: false, workspace: { isLocked: false } } })
+    MenuTrigger.mockImplementation(({ content }) => { return div([content]) })
+    TooltipTrigger.mockImplementation(({ content, children }) => {
+      const [open, setOpen] = useState(false)
+      return (div([
+        div(
+          {
+            onMouseEnter: () => {
+              setOpen(true)
+            },
+            onMouseLeave: () => {
+              setOpen(false)
+            }
+          },
+          [children]
+        ),
+        open && !!content && div([content])
+      ]))
+    })
+  })
+
+  it('renders Clone menu item as disabled', () => {
+    const onClone = jest.fn()
+    render(h(WorkspaceMenu, {
+      iconSize: 20, popupLocation: 'left',
+      callbacks: { onClone },
+      workspaceInfo: { name: 'example1', namespace: 'example-billing-project' }
+    }))
+    const menuText = screen.getByText('Clone')
+    expect(menuText).not.toHaveAttribute('disabled')
+  })
+
+  it('renders Clone tooltip', () => {
+    const onClone = jest.fn()
+    render(h(WorkspaceMenu, {
+      iconSize: 20, popupLocation: 'left',
+      callbacks: { onClone },
+      workspaceInfo: { name: 'example1', namespace: 'example-billing-project' }
+    }))
+    fireEvent.mouseOver(screen.getByText('Clone'))
+    expect(screen.queryByText('Cloning is not currently supported on Azure Workspaces')).toBeNull()
+  })
+
+  it('renders Share menu item as disabled', () => {
+    const onShare = jest.fn()
+    render(h(WorkspaceMenu, {
+      iconSize: 20, popupLocation: 'left',
+      callbacks: { onShare },
+      workspaceInfo: { name: 'example1', namespace: 'example-billing-project' }
+    }))
+    const menuText = screen.getByText('Share')
+    expect(menuText).toHaveAttribute('disabled')
+  })
+
+  it('renders Share menu item as enabled', () => {
+    useWorkspaceDetails.mockReturnValue({ workspace: { canShare: true, workspace: { isLocked: false } } })
+    const onShare = jest.fn()
+    render(h(WorkspaceMenu, {
+      iconSize: 20, popupLocation: 'left',
+      callbacks: { onShare },
+      workspaceInfo: { name: 'example1', namespace: 'example-billing-project' }
+    }))
+    const menuText = screen.getByText('Share')
+    expect(menuText).not.toHaveAttribute('disabled')
+  })
+
+  it('renders Share tooltip', () => {
+    const onShare = jest.fn()
+    render(h(WorkspaceMenu, {
+      iconSize: 20, popupLocation: 'left',
+      callbacks: { onShare },
+      workspaceInfo: { name: 'example1', namespace: 'example-billing-project' }
+    }))
+    fireEvent.mouseOver(screen.getByText('Share'))
+    screen.getByText('You have not been granted permission to share this workspace')
```

Comment:
Do you need to add an assertion that this isn't null?

Reply:
I didn't add the assertion that it isn't null because I thought getByText by default throws an error if it can't find a match (if it's null).

Reply:
Oh, I didn't notice `getByText` which is different form `queryByText`. I would find it easier to read/understand the code if you always used `queryByText` and verified that the result either is or is not null.

Reply:
yes, this was according to my guidance... getByText is a handy assert or throw where the failure will output super-handy DOM debug to the console.